File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int getHeight(void);
void buildPyramid(int current, int target);
void printXTimes(char character, int times);

int main(void)
{
    // Ask user for height
    int height = getHeight();
    // Build the double pyramid
    buildPyramid(1, height);
}

int getHeight(void)
{
    int height;
    do
    {
        // Ask user for integer input
        height = get_int("Height: ");
    }
    // Until value is more than 0 but less than 9
    while (height <= 0 || height > 8);
    return height;
}

void printXTimes(char character, int times)
{
    for (int i = 0; i < times; i++)
    {
        printf("%c", character);
    }
}

void buildPyramid(int current, int target)
{
    if (current > target)
    {
        return;
    }

    int whiteSpaces = target - current;

    // Print white spaces on the left hand side
    printXTimes(' ', whiteSpaces);
    // Print "blocks" on the left hand side
    printXTimes('#', current);
    // Print white spaces on the right hand side
    printXTimes(' ', 2);
    // Print "blocks" on the right hand side
    printXTimes('#', current);
    printf("\n");

    buildPyramid(current + 1, target);
}

File: sentimental-mario-more/mario.py
# Define get_int
def get_int(prompt: str):
    number: int = 0
    value = input(prompt)

    try:
        number = int(value)
    except:
        get_int(prompt)

    return number

# Define buildPyramid which will build the pyramid recursively


def buildPyramid(current: int, target: int):
    # Define the base case
    if (current > target):
        return 

    # Calculate total of columns
    whiteSpaces = target - current

    # Print white spaces on the left hand side
    for _ in range(whiteSpaces):
        print(' ', end='')
    # Print white spaces on the left hand side
    for _ in range(current):
        print('#', end='')
    # Print white spaces on the left hand side
    for _ in range(2):
        print(' ', end='')
    # Print white spaces on the left hand side
    for _ in range(current):
        print('#', end='')
    # Start a new line
    print('')
    
    # Call the function with updated arguments
    buildPyramid(current + 1, target)


while (True):
    # Ask the user for a height
    height = get_int("Height: ")
    if (height > 0 and height < 9):
        break

# Build and prompt the pyramid to the user
buildPyramid(1, height)
